lib/application/plus_button_settings/plus_button_settings_bloc.dart

```diff
@@ -37,7 +37,8 @@ class PlusButtonSettingsBloc
     Emitter<PlusButtonSettingsState> emit,
   ) async {
     emit(PlusButtonSettingsState.loadInProgress());
-    await _hubConnectionRepository.closeConnection();
-    context.router.replace(const ConnectToHubRoute());
+    await _hubConnectionRepository
+        .closeConnection()
+        .whenComplete(() => context.router.replace(const ConnectToHubRoute()));
```

Comment:
It's the same logic, there is no need to change it.
Also the await make sure that the lines are more readable and less indented forward

Reply:
I was trying to fix the warning that says to not use the build context across async gaps.